src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -2381,16 +2381,17 @@ static void read_action_locked(void* tp, grpc_error_handle error) {
   if (t->closed_with_error.ok()) {
     size_t i = 0;
     grpc_error_handle errors[3] = {error, absl::OkStatus(), absl::OkStatus()};
-    for (; i < t->read_buffer.count && errors[1] == absl::OkStatus(); i++) {
+
+    do {
```

Comment:
Commenting again for visibility: I think the test failures are due to this change, `read_buffer.count` may be 0. I recommend reverting this piece, and we can land the cleanup of the empty loop.

Reply:
Looks good, thanks for the contribution. Merged.